src/compiler/python_generator.cc

```diff
@@ -170,6 +170,45 @@ grpc::string ModuleAlias(const grpc::string& filename) {
   return module_name;
 }
 
+grpc::string ToSnakeCase(const grpc::string& method_name) {
+  grpc::string result;
+  size_t word_start = 0;
+  size_t word_end = 0;
+  while (true) {
+    word_start = word_end;
+    if (word_start + 1 < method_name.size() &&
+        std::isupper(method_name[word_start + 1])) {
+      // the word starting at `word_start` is an all-caps acronym
+      for (word_end = word_start + 1; word_end < method_name.size(); ++word_end) {
+        if (!std::isupper(method_name[word_end])) {
```

Comment:
What if we change this to ```if (std::islower())```.  Its not perfect, but I think it covers mixed caps/numerics slightly better.